### PR TITLE
feat: support removal of properties with --set name=undefined

### PR DESCRIPTION
**Pull request by jbedard**
WDYT of this one?

I've had a few cases where I need to remove properties. The common one being `jsc.paths` and `jsc.baseUrl` because they work differently then with tsc (swc rewrites imports to follow paths, typescript does not).

This adds `undefined` support, then `JSON.stringify` will remove the property. I'm not sure if swc has any properties where `null` is a valid value, but if so that would be another minor addition to the `parseParamValue` method.

The other option would be `--del jsc.paths` instead of using `undefined`, then we don't have to make the `parseParamValue` more complicated? But more cli params makes the CLI more complicated, and would not solve the `null` case if that is ever needed 🤷

### CODE REVIEW

File: src/utils.ts [hunk 1]
```diff
@@ -76,6 +76,9 @@ export function parseParamValue(prop: string, value: string) {
 	if (value === 'true' || value === 'false') {
 		return value === 'true'
 	}
+	if (value === 'undefined') {
+		return undefined
```
Comment (songkeys) [inline src/utils.ts:77]: > This adds undefined support, then JSON.stringify will remove the property. I'm not sure if swc has any properties where null is a valid value

Perhaps `delete` operator is a better way?

Comment (jbedard) [reply src/utils.ts:77]: Maybe that is better to minimize the number of odd cases in this method? Although I would vote for `--unset`, the opposite of `--set`... 🤔 

If we do find more odd cases required here... I might vote to replace `parseParamValue` with just `JSON.parse`. Then it would support true/false/numbers/null/objects/array etc, without any magic cases like this method. It just means it's a bit of a breaking change because you'll now have to quote all string values...

Comment (songkeys) [reply src/utils.ts:77]: `--unset` introcudes more complexity.

And if replacing `parseParamValue` with just `JSON.parse`, to handle strings - a) try catch, if in catch, that's probably a string; b) user have to pass string with quotes`''`. I prefer the option a cuz it introduces less mental burden to users.

---

I was also thinking of changing this line https://github.com/Songkeys/tsconfig-to-swcconfig/blob/1807dcdc84179019ff373254e18e48e43813cfd1/src/cli.ts#L66C16-L66C16

```diff
-	parent[key] = parseParamValue(prop, value)
+	parseParamValue(parent, key)
```

and the `parseParamValue` func will mutate the `parent[key]` so that we can also do `delete parent[key]`. This is to solve the unnecessary `undefined`-field problem.

Comment (jbedard) [reply src/utils.ts:77]: Oh, I see what you mean, I thought you meant a `--delete` cli param which I agree does add complexity.

Yeah the try/catch to avoid quotes 99% of the time is worth removing that mental burden. What represents a "delete" action though? We could say `null` does the delete, but maybe there are swc options where `null` is valid. We could say the `undefined` string does the delete even though `JSON.parse(undefined)` throws.

Basically...
```
if ("undefined" === value) delete parent[key]
else {
  try {
    parent[key] = JSON.parse(value)
  } catch (_) {
    parent[key] = value
  }
}
```
?

The only thing I can think of is if that exception was legit the user now has no warning that it is being treated as a plain string. I guess they can just look at the output to figure that out though 👍

Comment (songkeys) [reply src/utils.ts:77]: Yes that's what I meant!

We might not want to output warnings so far cuz some user might manipulate the output to stdout directly. Let's keep things simple first.

Comment (jbedard) [reply src/utils.ts:77]: Ok, using `JSON.parse` made this a lot nicer.

However we can't use `delete` because the object being returned here gets "merged" into the converted result, so the `undefined` value needs to be in this object so it can later overwrite/merge into the main one. I think that makes sense though?

Comment (songkeys) [reply src/utils.ts:77]: Ah yes. I forgot about this.

**Comment by songkeys** [comment]
Thank you!
